src/lib/controllers/sso_sp.php

```diff
@@ -1,56 +1,59 @@
 <?php
   // Setup Default Options Array
-  
-  if (isset($_POST['submit']) && wp_verify_nonce($_POST['_wpnonce'],'sso_sp') ) 
-  {    
+
+  if (isset($_POST['submit']) && wp_verify_nonce($_POST['_wpnonce'],'sso_sp') )
+  {
+      $upload_dir = constant('SAMLAUTH_CONF') . '/certs/' . get_current_blog_id();
```

Comment:
@ktbartholomew Sorry, actually, after looking at it, it seems simple saml does not accept any key data, it only accepts the file path as an argument (https://simplesamlphp.org/docs/stable/simplesamlphp-reference-sp-remote). So this has to be retained if the private key changes.

I have also added the hack at https://github.com/ktbartholomew/saml-20-single-sign-on/pull/5/files#diff-09c345e6a270ffcaebbe3a83ca9ec990R85 in case the container was rebuilt and the file does not exist yet.